jac-uk/admin
# Search Improvements - Digital Platform 875

## What's included?

 Add a new empty state pattern.
 Ensure the search works when the user types 3 or more charaters in the search input.
 When performing a search if a valid search term has been passed in then override pagination to use the 'number' format. If the search term is removed or invalid then revert it back to whatever the initial value was.

This PR is linked to issue: https://github.com/jac-uk/digital-platform/issues/875
And PR: https://github.com/jac-uk/digital-platform/pull/920

Closes https://github.com/jac-uk/digital-platform/issues/875

## Who should test?
✅ Product owner
✅ Developers
✅ UTG

## How to test?

When no search results are available there is a new 'empty search results' screen.
See screenshot:
<img width="1224" alt="Screenshot 2023-08-08 at 14 28 33" src="https://github.com/jac-uk/admin/assets/115651787/d7c46e52-e216-4dd4-ba2b-47af3bfea353">

The new search has been implemented on six different search pages.
Each page will have different criteria that can be searched on.
The search requires three characters or more to be typed, ie if you type 'ja' then the search will not execute but if you type 'jac' then it will.
**Be sure to also test the search using the @ symbol and the dot character (if it appears in the email address).**

The list below gives the url of the search page and the criteria that can be searched on:

- Exercises
   See: https://jac-admin-develop--pr2105-feature-dp-875-searc-sloajkfu.web.app/exercises
   Search Criteria: Exercise Title, Exercise Reference Number
- Candidates
   See: https://jac-admin-develop--pr2105-feature-dp-875-searc-sloajkfu.web.app/candidates/
   Search Criteria: Candidate Name, Candidate Email, Candidate NI Number
- Applications
   See: https://jac-admin-develop--pr2105-feature-dp-875-searc-sloajkfu.web.app/exercise/mgiIdf4ccm3z7Q3jsZKb/applications/applied
   Search Criteria: Application Reference Number,  Candidate Name, Candidate Email, Candidate NI Number
- Character Checks
   See: https://jac-admin-develop--pr2105-feature-dp-875-searc-sloajkfu.web.app/exercise/mgiIdf4ccm3z7Q3jsZKb/tasks/character-checks
   Search Criteria: Application Reference Number,  Candidate Name, Candidate Email, Candidate NI Number
- Assessments
   See: https://jac-admin-develop--pr2105-feature-dp-875-searc-sloajkfu.web.app/exercise/mgiIdf4ccm3z7Q3jsZKb/tasks/independent-assessments
   Search Criteria: Application Reference Number,  Candidate Name, Assessor Email, Assessor Name
- Application Records
   See: https://jac-admin-develop--pr2105-feature-dp-875-searc-sloajkfu.web.app/exercise/mgiIdf4ccm3z7Q3jsZKb/stages/review
   Search Criteria: Application Reference Number,  Candidate Name, Candidate Email, Candidate NI Number


## Risk - how likely is this to impact other areas?
🟢 No risk - this is a self-contained piece of work
🟠 Medium risk - this does change code that is shared with other areas
🔴 High risk - this includes a lot of changes to shared code

## Related permissions
Have permissions been considered for this functionality?
- No permission changes required

---
PREVIEW:DEVELOP
_can be OFF, DEVELOP or STAGING_
